src/hotspot/share/opto/loopnode.cpp

```diff
@@ -802,14 +802,14 @@ bool PhaseIdealLoop::create_loop_nest(IdealLoopTree* loop, Node_List &old_new) {
   // The number of iterations for the integer count loop: guarantee no
   // overflow: max_jint - stride_con max. -1 so there's no need for a
   // loop limit check if the exit test is <= or >=.
-  int iters_limit = max_jint - ABS(stride_con) - 1;
+  int iters_limit = max_jint - uabs(stride_con) - 1;
 #ifdef ASSERT
   if (bt == T_LONG && StressLongCountedLoop > 0) {
     iters_limit = iters_limit / StressLongCountedLoop;
   }
 #endif
   // At least 2 iterations so counted loop construction doesn't fail
-  if (iters_limit/ABS(stride_con) < 2) {
+  if (iters_limit/uabs(stride_con) < 2) {
```

Comment:
Previously negative values of iters_limit would cause us to return false here.  Now the comparison is unsigned, so we can get a different result:
stride_con is max_jint
iters_limit is -1
(unsigned) -1 / max_jint is 2
so we no longer return false here.
Using uabs with signed values is error-prone.